### PR TITLE
update readme to be more comprehensive

### PR DESCRIPTION
**Pull request by DavidVorick**

### CODE REVIEW

**Comment by lukechampine** [comment]
I would also add references to the [Fortuna](https://www.schneier.com/academic/paperfiles/fortuna.pdf) paper and point out that FreeBSD uses Fortuna as its `/dev/random`. I think the main thing that spooks people is that they believe `/dev/urandom` is doing something deeply complex that can't be implemented in userspace.

**Comment by DavidVorick** [comment]
Updated to match the changes in my other PR. The benchmarks were performed while my computer was under load, and are therefore not the best. Would be good if @lukechampine can run the benchmarks on his computer and add a commit to update them appropriately.

File: README.md [hunk 1]
```diff
@@ -8,18 +8,49 @@ fastrand
...
+BenchmarkRead512K-4                   	    1000	   1403510 ns/op	 364.80 MB/s
+BenchmarkReadCrypto512K-4             	      50	  39849581 ns/op	  12.85 MB/s
+
+// 32 byte reads using 4 threads
+BenchmarkRead4Threads-4               	 3000000	       372 ns/op	 343.35 MB/s
+BenchmarkReadCrypto4Threads32-4       	  100000	     16060 ns/op	   7.97 MB/s
+
+// 512 kb reads using 4 threads
+BenchmarkRead4Threads512k-4           	     500	   2417701 ns/op	 847.09 MB/s
+BenchmarkReadCrypto4Threads512kb-4    	      10	 224889487 ns/op	   9.11 MB/s
```
Comment (lukechampine) [inline README.md]: my benchmarks:
```
BenchmarkRead32-4                     	10000000	       175 ns/op	 181.86 MB/s
BenchmarkReadCrypto32-4               	  500000	      2733 ns/op	  11.71 MB/s

BenchmarkRead512K-4                   	    1000	   1336217 ns/op	 383.17 MB/s
BenchmarkReadCrypto512K-4             	      50	  33423693 ns/op	  15.32 MB/s

BenchmarkRead4Threads-4               	 3000000	       392 ns/op	 326.46 MB/s
BenchmarkReadCrypto4Threads32-4       	  200000	      7579 ns/op	  16.89 MB/s

BenchmarkRead4Threads512k-4           	    1000	   1899048 ns/op	1078.43 MB/s
BenchmarkReadCrypto4Threads512kb-4    	      20	  97423380 ns/op	  21.02 MB/s
```

**Comment by DavidVorick** [comment]
Updated. Tried running math/rand in parallel (add benchmarks for it as well) and, surprise surprise math/rand actually slows down significantly when called from multiple threads.

So, fastrand will actually outperform math/rand for multithreaded situations.

**Pull request merged**